### DISCUSSION TITLE
(PDOException(code: 20002): SQLSTATE[01002] Adaptive Server connection failed

### DISCUSSION THREAD

**Discussion by Xanyone** [laravel/framework#47711] [Q&A]
### Laravel Version

10.4.1

### PHP Version

8.2.8

### Database Driver & Version

SQLSERVER2014 for windows server 2016 datacenter

### Description

This machine is macOS Ventura 13.14.1 and was previously able to connect to it. However, after updating the brew package, I have been constantly prompted for this issue. I have tried many solutions, but they still do not work

### Steps To Reproduce

[previous exception] [object] (PDOException(code: 20002): SQLSTATE[01002] Adaptive Server connection failed (x.x.x.x) (severity 9)

**Comment by ilcors-dev** [comment]
i'm having the same problem both on a ubuntu & macos machine, both running php 8.2.8

## Laravel Version
10.18.0

## PHP Version
8.2.8

## Database Driver & Version
sqlsrv & pdo_sqlsrv version 5.11.0, SQLSERVER 2019 Standard (10.0)

## Description
The behaviour is kinda strange, i'm able to navigate and use the application normally via browser but when trying to test it via PEST a PDOException gets thrown.
The application is running locally in a docker container using sail, and the same happens when running ```./vendor/bin/pest``` from the container terminal (accessed via the command ```sail shell```)

Below the stack trace with the exception message
```
previous: PDOException {#1517
    #message: "SQLSTATE[01002] Adaptive Server connection failed (x.x.x.x) (severity 9)"
    #code: 20002
    #file: "./vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php"
    #line: 65
    +errorInfo: null
    trace: {
      ./vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php:65 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php:83 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php:48 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connectors/SqlServerConnector.php:32 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php:184 { …}
      Illuminate\Database\Connectors\ConnectionFactory->Illuminate\Database\Connectors\{closure}() {}
      ./vendor/laravel/framework/src/Illuminate/Database/Connection.php:1275 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connection.php:1311 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connection.php:521 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connection.php:416 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connection.php:788 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connection.php:934 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connection.php:913 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connection.php:757 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Connection.php:407 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2747 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2736 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:3290 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2735 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:720 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:704 { …}
      ./vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:941 { …}
      ./vendor/laravel/framework/src/Illuminate/Support/Traits/ForwardsCalls.php:23 { …}
      ./vendor/spatie/laravel-query-builder/src/QueryBuilder.php:105 { …}
      ./app/Actions/Article/ListArticles.php:34 {
        App\Actions\Article\ListArticles->handle(ListArticlesDataRequest $request): PaginatedDataCollection^
        › ->with($request->with ?? [])
        › ->simplePaginate()
        › ->appends(request()->query());
        arguments: {
          $name: "simplePaginate"
          $arguments: []
        }
      }
... continues
```

My ```database.php``` configuration for the sqlsrv connection
```
'sqlsrv' => [
    'driver' => 'sqlsrv',
    'url' => env('DATABASE_URL'),
    'host' => env('DB_HOST', 'localhost'),
    'port' => env('DB_PORT', '1433'),
    'database' => env('DB_DATABASE', 'forge'),
    'username' => env('DB_USERNAME', 'forge'),
    'password' => env('DB_PASSWORD', ''),
    'charset' => 'utf8',
    'prefix' => '',
    'prefix_indexes' => true,
    'encrypt' => env('DB_ENCRYPT', 'yes'),
    'trust_server_certificate' => env('DB_TRUST_SERVER_CERTIFICATE', 'true'),
],
```

## Steps To Reproduce
- laravel application with php 8.1.22 / 8.2.8 and Microsoft pdo driver, either macos or linux,
- run command ```./vendor/bin/pest```

**Comment by ilcors-dev** [comment]
I also tried lowering the php version to ```php8.1.22```, but the same happens

**Comment by ilcors-dev** [comment]
Okay, found the culprit.
I'll write down the solution so anyone running into the same issue will solve it swiftly.
 
Apparently the pdo_sqlsrv & sqlsrv drivers **were not correctly installed**, both on my ubuntu machine (which is using php-fpm) and locally on my macos machine (plain php installed view homebrew)
To solve the issue i had to install the drivers following the official microsoft's docs 
- [for the pdo_sqlsrv & sqlsrv drivers](https://learn.microsoft.com/en-us/sql/connect/php/installation-tutorial-linux-mac?view=sql-server-ver16)
- [for the ODBC driver](https://learn.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-driver-manager?view=sql-server-ver15)

after that, I was able to run my tests via pest without problems!
